client/packages/common/src/ui/layout/tables/components/Cells/MultiNumberInputCell/MultiNumberInputCell.tsx

```diff
@@ -0,0 +1,167 @@
+ * ratio of 1). For example, for years and months we would define units like so:
+ *
+ * units={[
+ *         {
+ *           key: 'year',
+ *           ratio: 12,
+ *           label: "Y"
+ *         },
+ *         {
+ *           key: 'month',
+ *           ratio: 1,
+ *           label: "M"
+ *         },
+ *    ]}
+ *
+ * For hours/minutes/seconds:
+ *
+ * units={[
+ *         {
+ *           key: 'hour',
+ *           ratio: 3600,
+ *           label: "H"
+ *         },
+ *         {
+ *           key: 'minute',
+ *           ratio: 60,
+ *           label: "M"
+ *         },
+ *         {
+ *           key: 'second',
+ *           ratio: 1,
+ *           label: "S"
+ *         },
+ *    ]}
+ */
+
+import React from 'react';
+import { CellProps } from '../../../columns';
+import {
+  NumericInputProps,
+  NumericTextInput,
+  StandardTextFieldProps,
+} from '@common/components';
+import { Box } from '@openmsupply-client/common';
+import { RecordWithId } from '@common/types';
+import { useBufferState, useDebounceCallback } from '@common/hooks';
+
+interface Unit {
+  key: string;
+  ratio: number;
+  label: string;
+}
+
+export const MultipleNumberInputCell = <T extends RecordWithId>({
+  units,
+  rowData,
+  column,
+  rowIndex,
+  columnIndex,
+  isDisabled = false,
+  min = 0,
+  max,
+  decimalLimit,
+  step,
+  multiplier,
+  defaultValue,
+  allowNegative,
+  id,
+  TextInputProps,
+  width,
+}: CellProps<T> &
+  NumericInputProps & {
+    id?: string;
+    TextInputProps?: StandardTextFieldProps;
+  } & { units: Unit[] }): React.ReactElement<CellProps<T>> => {
+  const [buffer, setBuffer] = useBufferState(column.accessor({ rowData }));
+  const updater = useDebounceCallback(column.setter, [column.setter], 350);
+
+  const autoFocus = rowIndex === 0 && columnIndex === 0;
+
+  const cellValues = getCellValues(buffer as number | undefined, units);
+
+  return (
+    <Box display="flex">
+      {units.map((unit, index) => (
+        <NumericTextInput
+          key={unit.key}
+          id={id}
+          disabled={isDisabled}
+          autoFocus={autoFocus}
+          {...TextInputProps}
+          InputProps={{
+            sx: { '& .MuiInput-input': { textAlign: 'right' } },
+            ...TextInputProps?.InputProps,
+          }}
+          onChange={num => {
+            const newValue = num === undefined ? min : num;
+            if (cellValues[index] === newValue) return;
+            const newValues = [...cellValues];
+            newValues[index] = newValue;
+            const newTotal = computeTotal(newValues, units);
+            setBuffer(newTotal);
+            updater({ ...rowData, [column.key]: Number(newTotal) });
+          }}
+          min={min}
+          // Limit smaller units to less than the value of the next highest
+          // unit, e.g. "Months" would be limited to 11.9999
+          max={setMax(units, index, max)}
+          decimalLimit={decimalLimit}
```

Comment:
Do you think we could require all the earlier units to be integers, and only the last accept decimals? This seems really unintuitive to me:


https://github.com/user-attachments/assets/96f14f08-0b5d-4a08-a97a-a23c08a23720

Reply:
Yes, for sure, there's no reason to ever need decimals for anything other than the smallest unit.

Reply:
Yes, for sure, there's no reason to ever need decimals for anything other than the smallest unit.